RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -6105,3 +6105,43 @@ float mp_regen()
 {
 	return 0.5 * (numeric_modifier("MP Regen Min") + numeric_modifier("MP Regen Max"));
 }
+
+boolean auto_canForceNextNoncombat()
+{
+	return auto_pillKeeperAvailable()
+	|| (!get_property("_claraBellUsed").to_boolean() && (item_amount($item[Clara\'s Bell]) > 0))
+	|| (item_amount($item[stench jelly]) > 0 && auto_is_valid($item[stench jelly]) && spleen_left() < $item[stench jelly].spleen);
+}
+
+boolean _auto_forceNextNoncombat()
+{
+	// Use stench jelly or other item to set the combat rate to zero until the next noncombat.
+	// There's no way of knowing if we've already used one, so the caller needs to be careful.
```

Comment:
it may be worth increasing complexity a little bit and introduce some tracking properties. seems like this is begging for wasted uses and confused state. especially with the prevalence of the ioexceptions these days. Maybe an `auto_forceNonCombatSource` and set it to none, clara, stench, pill accordingly. Of course that still has room for error if we dont properly reset that property (presumably in auto_post_adv.ash). Maybe track the adventure count it was used on as well, then we could at least try to do some deduced error recovery if we've adventured some number of times since we forced NC we can probably assume we triggered the NC and can clear the properties...

Idk might be more trouble than its worth, any other ways you can think of making this a little less "caller beware"?

Reply:
This has some potential footguns . I'm not sure how to robustly check that we've hit an NC at all in `auto_post_adv.ash`. Checking that the `lastEncounter` property doesn't parse to a monster probably does 90% of what we want, so I'll try that.

Reply:
yes, foot guns abound. 90% is still better than purely relying on caller. we can always improve from there if we find its a problem.